ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs

```diff
@@ -0,0 +1,346 @@
+                   -> (NtNAddr, RelayAccessPoint)
+                   -> Gen SimArgs
+      addressToRun raps dMap (ntnAddr, rap) = do
+        bgaSlotDuration <- fromInteger <$> choose (0, 100)
+        bgaSeed <- mkStdGen <$> arbitrary
+        quota <- chooseInt (0, 100)
+
+        -- These values approximately correspond to false positive
+        -- thresholds for streaks of empty slots with 99% probability,
+        -- 99.9% probability up to 99.999% probability.
+        -- t = T_s [log (1-Y) / log (1-f)]
+        -- Y = [0.99, 0.999...]
+        --
+        -- T_s = slot length of 1s.
+        -- f = 0.05
+        -- The timeout is randomly picked per bearer to avoid all bearers
+        -- going down at the same time in case of a long streak of empty
+        -- slots. TODO: workaround until peer selection governor.
+        -- Taken from ouroboros-consensus/src/Ouroboros/Consensus/Node.hs
+        mustReplyTimeout <- Just <$> oneof (pure <$> [90, 135, 180, 224, 269])
+
+        stdGen <- mkStdGen <$> arbitrary
+
+        lrp <- genLocalRootPeers raps rap
+
+        peerSelectionTargets <- arbitrary
+        dnsTimeout <- arbitrary
+        dnsLookupDelay <- arbitrary
+
+        return ( bgaSlotDuration
+               , bgaSeed
+               , quota
+               , mustReplyTimeout
+               , raps
+               , stdGen
+               , dMap
+               , ntnAddr
+               , lrp
+               , peerSelectionTargets
+               , dnsTimeout
+               , dnsLookupDelay
+               )
+
+-- | Run an arbitrary topology
+multinodeSim :: forall m. ( MonadAsync m
+                         , MonadFork m
+                         , MonadST m
+                         , MonadEvaluate m
+                         , MonadLabelledSTM m
+                         , MonadCatch       m
+                         , MonadMask        m
+                         , MonadTime        m
+                         , MonadTimer       m
+                         , MonadThrow  (STM m)
+                         , Eq (Async m Void)
+                         , forall a. Semigroup a => Semigroup (m a)
+                         )
+                      => BearerInfo
+                      -> MultiNodeScript
+                      -> m Void
+multinodeSim
+  defaultBearerInfo
+  (MultiNodeScript args) = withAsyncAll (map runNode args) $ \nodes -> do
+    (_, x) <- waitAny nodes
+    return x
+  where
+    runNode :: SimArgs -> m Void
+    runNode ( bgaSlotDuration
+            , bgaSeed
+            , quota
+            , mustReplyTimeout
+            , raps
+            , stdGen
+            , dMap
+            , rap
+            , lrp
+            , peerSelectionTargets
+            , dnsTimeout
+            , dnsLookupDelay
+            ) =
+      withSnocket nullTracer defaultBearerInfo Map.empty
+        $ \ntnSnocket _ -> withSnocket nullTracer defaultBearerInfo Map.empty
+        $ \ntcSnocket _ ->
```

Comment:
I wonder if this will work:
```suggestion
      withSnocket nullTracer defaultBearerInfo Map.empty
        $ \ntnSnocket _ ->
          withSnocket nullTracer defaultBearerInfo Map.empty
        $ \ntcSnocket _ ->
```
anyway splitting the line is a good idea, even if the later one needs more indentation.